src/core/ui.js

```diff
@@ -79,6 +81,19 @@ document.documentElement.addEventListener("click", () => {
 });
 respecUI.appendChild(menu);
 
+menu.addEventListener("keydown", e => {
```

Comment:
This still seems out of place. This should all be handled by the menu closing (i.e., we should already have much of this code here, or it should be part of ui.hide() or .close() (whichever it is below... on my phone so hard to check)